libs/langchain/tests/unit_tests/test_imports.py

```diff
@@ -60,3 +62,103 @@ def test_import_all_using_dir() -> None:
                 continue
             # Attempt to import the name from the module
             getattr(mod, name)
+
+
+# Quick check to verify that folks aren't modifying the __all__ variable to add
+# imports to langchain_community
+def test_no_more_changes_to_proxy_community() -> None:
+    """This test is meant to catch any changes to the proxy community module.
+
+    Imports from langchain to community are officially DEPRECATED. Contributors
+    should not be adding new imports from langchain to community. This test
+    is meant to catch any new changes to the proxy community module.
+    """
+    library_code = PKG_ROOT / "langchain"
+    hash_ = 0
+    for path in library_code.rglob("*.py"):
+        # Calculate the relative path to the module
+        if not str(path).endswith("__init__.py"):
+            continue
+
+        deprecated_lookup = extract_deprecated_lookup(str(path))
+        if deprecated_lookup is None:
+            continue
+
+        # This uses a very simple hash, so it's not foolproof, but it should catch
+        # most cases.
+        hash_ += len(str(sorted(deprecated_lookup.items())))
+
+    evil_magic_number = 38620
```

Comment:
are we sure evil magic number is better than just dumping all the current lookup dicts somewhere 🌝

Reply:
never been sure of anything more in my life -- less work and easy to catch modifications

Reply:
easier to understand if the actual dicts are there, and you've already got all the functionality in the tests to make a script to extract current dicts — but i defer to ya on this one 👍